### ISSUE TITLE
[BUG] Settings are not saving when changed via the launcher.

### ISSUE THREAD

**Issue by hakimzulkufli** [jstaf/onedriver#364]
# System Information
openSUSE Tumbleweed Linux 6.5.9 (64-bit)
onedriver v0.14.1

# Problem

Hello,

When configuring OneDriver via the GUI launcher, settings were not saved. The launcher does not even attempt to save the config file in our home directory.

Here is a log of me changing the log level:

```
user@localhost:~> onedriver-launcher 

{"level":"warn","error":"open /home/user/.config/onedriver/config.yml: no such file or directory","path":"/home/user/.config/onedriver/config.yml","time":"2023-11-03T22:05:01+08:00","message":"Configuration file not found, using defaults."}

22:05:01 INF onedriver-launcher v0.14.1 15a1c642
22:05:01 INF Found existing mount. mount=/home/user/OneDrive
22:08:37 DBG Log level changed. newLevel=info
22:08:39 DBG Log level changed. newLevel=warn
```

The next run:
```
user@localhost:~> onedriver-launcher 

{"level":"warn","error":"open /home/user/.config/onedriver/config.yml: no such file or directory","path":"/home/user/.config/onedriver/config.yml","time":"2023-11-03T22:08:58+08:00","message":"Configuration file not found, using defaults."}

22:08:58 INF onedriver-launcher v0.14.1 15a1c642
```

As you can see above, the launcher still cannot find the config file because it wasn't created.

However, if I create the config file manually at `~/.config/onedriver/config.yml`, the settings were saved as expected. I assume the launcher failed to save the config file because it could not or did not create the directory. So I tested it by removing the config file while keeping the directory and the launcher is able to save the config file successfully.

# Summary

The GUI application `onedriver-launcher` does not create the directory `~/.config/onedriver` at first launch, causing it to not be able to save the configuration file.

Ps. I've been using onedriver since 2020. One thing I've noticed is how "invisible" onedriver is to the internet search engines. On Google, searching for `onedriver` or `onedrive linux` returns nothing about onedriver at all. onedriver has the best implementation as it behaves exactly as how it is in the official Windows client.

I hope one day you will get the recognition you deserve and more OneDrive users will use your driver. Thank you very much for your amazing work, jstaf.

**Comment by jstaf** [comment]
oops, you're right! fixed in #365.

And thank you :heart:! Yeah I have never advertised this software ever actually (in any form), people just sort of found it early on and now I have users :smiling_face_with_tear:. I am somewhat okay with the low profile because this was originally a for-fun project. It's both nice (yay, other people appreciate things!), and also kind of limiting because I feel like I've got an obligation to maintain things going forwards. "C'est la vie" I guess haha

**Comment by hakimzulkufli** [comment]
Thanks! I've confirmed that the launcher is now working properly.

And I understand your feelings very well. onedriver is already perfect for me. Just having it at its current scope it already great. So thank you again for the amazing work.
